tests/conftest.py

```diff
@@ -185,3 +186,40 @@ def custom_template(tmp_dir, dvc):
 @pytest.fixture(autouse=True)
 def mocked_webbrowser_open(mocker):
     mocker.patch("webbrowser.open")
+
+
+@pytest.fixture(autouse=True)
+def isolate(tmp_path_factory, monkeypatch) -> None:
+    path = tmp_path_factory.mktemp("mock")
+    home_dir = path / "home"
+    home_dir.mkdir()
+
+    if sys.platform == "win32":
+        home_drive, home_path = os.path.splitdrive(home_dir)
+        monkeypatch.setenv("USERPROFILE", str(home_dir))
+        monkeypatch.setenv("HOMEDRIVE", home_drive)
+        monkeypatch.setenv("HOMEPATH", home_path)
+
+        for env_var, sub_path in (
+            ("APPDATA", "Roaming"),
+            ("LOCALAPPDATA", "Local"),
+        ):
+            path = home_dir / "AppData" / sub_path
+            path.mkdir(parents=True)
+            monkeypatch.setenv(env_var, os.fspath(path))
+    else:
+        monkeypatch.setenv("HOME", str(home_dir))
+
+    monkeypatch.setenv("GIT_CONFIG_NOSYSTEM", "1")
+    contents = b"""
+[user]
+name=DVC Tester
+email=dvctester@example.com
+[init]
+defaultBranch=master
```

Comment:
Isn't this supposed to be `main`?

Reply:
This is for tests, we always assumed it to be master there. There's https://github.com/iterative/dvc/issues/7005 open to address that.